includes/class-wp-job-manager-shortcodes.php

```diff
@@ -360,10 +360,24 @@ public function output_jobs( $atts ) {
 
 		// Array handling.
 		$atts['categories']         = is_array( $atts['categories'] ) ? $atts['categories'] : array_filter( array_map( 'trim', explode( ',', $atts['categories'] ) ) );
+		$atts['selected_category']  = is_array( $atts['selected_category'] ) ? $atts['selected_category'] : array_filter( array_map( 'trim', explode( ',', $atts['selected_category'] ) ) );
 		$atts['job_types']          = is_array( $atts['job_types'] ) ? $atts['job_types'] : array_filter( array_map( 'trim', explode( ',', $atts['job_types'] ) ) );
 		$atts['post_status']        = is_array( $atts['post_status'] ) ? $atts['post_status'] : array_filter( array_map( 'trim', explode( ',', $atts['post_status'] ) ) );
 		$atts['selected_job_types'] = is_array( $atts['selected_job_types'] ) ? $atts['selected_job_types'] : array_filter( array_map( 'trim', explode( ',', $atts['selected_job_types'] ) ) );
 
+		// Normalize field for categories.
```

Comment:
I wonder if it would make sense to move this functionality into `get_job_listings`, which this ultimately calls, to make this functionality more widely available? For example, I could see it being useful in the REST API in the future?

Reply:
This doesn't actually get passed directly to `get_job_listings`, but just pre-selects the categories on the search form. Do you think we should change the behavior to override the search form and use the query string when performing the actual search?

Reply:
Ah I think I see my confusion. It gets passed directly in the non-ajax case (see line 425). So I was thinking that we could just do this handling inside get_job_listings for the attributes handed in.

But I forgot about the ajax case. For that, we would need to do it this way in order to pre-populate the field, as you said. So probably this solution is best for now. If we want to add this handling more generically later, we can deal with it then 🙂